synapse/models/files.py

```diff
@@ -699,6 +702,42 @@ def getModelDefs(self):
                         'doc': 'The file offset to the beginning of the section'}),
                 )),
 
+                ('file:mime:lnk', {}, (
+                    ('flags', ('int', {}), {
+                        'doc': 'The flags specified by the LNK header that control the structure of the LNK file.'}),
+                    ('entry:primary', ('file:path', {}), {
+                        'doc': 'The primary file path contained within the FileEntry structure of the LNK file.'}),
+                    ('entry:secondary', ('file:path', {}), {
+                        'doc': 'The secondary file path contained within the FileEntry structure of the LNK file.'}),
+                    ('entry:extended', ('file:path', {}), {
+                        'doc': 'The extended file path contained within the extended FileEntry structure of the LNK file.'}),
+                    ('entry:localized', ('file:path', {}), {
+                        'doc': 'The localized file path contained within the extended FileEntry structure of the LNK file.'}),
+                    ('entry:icon', ('file:path', {}), {
+                        'doc': 'The icon file path contained within the StringData structure of the LNK file.'}),
+                    ('environment:path', ('file:path', {}), {
+                        'doc': 'The target file path contained within the EnvironmentVariableDataBlock structure of the LNK file.'}),
+                    ('environment:icon', ('file:path', {}), {
+                        'doc': 'The icon file path contained within the IconEnvironmentDataBlock structure of the LNK file.'}),
+                    ('working', ('file:path', {}), {
+                        'doc': 'The working directory used when activating the link target.'}),
+                    ('relative', ('str', {}), {
+                        'doc': 'The relative target path string contained within the StringData structure of the LNK file.'}),
+                    ('arguments', ('it:cmd', {}), {
+                        'doc': 'The command line arguments passed to the target file when the LNK file is activated.'}),
+                    ('desc', ('str', {}), {
```

Comment:
Do we need to strip whitespace off this value?
Should it have a text display hint?

Reply:
display hint yes.

Stripping whitespace I'm a bit more worried about since this is coming out of a section that specifically disallows null terminators (you extract this field purely by the count of characters from a prepended integer), so extra whitespace at the end would be interesting to see and could differentiate things.

Reply:
We definitely want this data to be raw since it's coming from a header :+1: